dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py

```diff
@@ -503,6 +503,64 @@ def test_merge_num_zeros_and_negatives(self):
         self.assertEqual(num_profiler.num_zeros, 0)
         self.assertEqual(num_profiler.num_negatives, 0)
 
+    def test_profile(self):
+        num_profiler = TestColumn()
+
+        mock_profile = dict(
+            min=1.0,
+            max=1.0,
+            sum=1.0,
+            mean=0, # default
+            variance=np.nan, # default
+            skewness=np.nan, # default
+            kurtosis=np.nan, # default
+            stddev=np.nan, # default
+            histogram={
+                'bin_counts': np.array([1, 1, 1]),
+                'bin_edges': np.array([1.0, 2.0, 3.0, 4.0])
+            },
+            quantiles={
+                0: 2.0,
+                1: 3.0,
+                2: 4.0,
+            },
+            num_zeros=0, # default
+            num_negatives=0, # default
+            times=defaultdict(float), # default
+        )
+
+        num_profiler.match_count = 0
+        num_profiler.min = mock_profile['min']
+        num_profiler.max = mock_profile['max']
+        num_profiler.sum = mock_profile['sum']
+        num_profiler.histogram_selection = 'auto'
+        num_profiler.histogram_methods['auto']['histogram'] = \
+            mock_profile['histogram']
+        num_profiler.quantiles = mock_profile['quantiles']
+        num_profiler.times = mock_profile['times']
```

Comment:
this seems strange to me that we are manually setting the values and just checking if the profile returns correctly.

I would rather send actual data into the profiler and ensure the calcs / results out are correct.

Reply:
To test if the profile works, I think assigning properties and checking them later is not bad. At least, we did that for other tests for numeric profile as well, using TestColumn(). If we want to send the data, we have to run it under int/float/text.